### ISSUE TITLE
cam_chempp issue with usr_mech_infile when path to compiler is too long

### ISSUE THREAD

**Issue by annlew** [NorESMhub/CAM#160]
### Issue Type

Infrastructure Update

### Issue Description

When using user provided chemistry mechanism file for a case, the build fails if the path to cpp is longer than 64 characters. Character length is hardcoded in: 

components/src/cam_chempp/mozpp.main.f

      `character(len=64)  :: cpp_dir, cpp_opts`

This turned out to be an issue on Tetralith after OS upgrade and updated compiler suite when using tag cam_cesm2_1_rel_05-Nor_v1.0.5.

This is easy to fix by increasing char length for cpp_dir and I can create a pull request to fix this. However, I'm not sure which branch this should go into in that case. Seems like the structure for the chemistry preprocessor has changed in more recent versions. 

### Will this change answers?

No

### Will you be implementing this yourself?

Yes

**Comment by gold2718** [comment]
@annlew, Thanks for taking this on! I have a few notes / questions:

- Are you targeting the noresm2_1_develop branch, the noresm2_3_develop branch, or both with this issue?
- What length do we want to move towards?
    - One possibility is `len=256` since that is a standard length we use in CAM for files (`SHR_KIND_CL`).
    - Another is `len=320` since that seems to be used for some paths in the same program.
    - There are a lot of path-like variables. Should we create a parameter and use that for all of these variables?
- The filepath in the preamble above does not seem right. I think `<root>/components/cam/chem_proc/src/cam_chempp/mozpp.main.f` or `<root>/src/cam_chempp/mozpp.main.f` would be more clear.
- This will provide a fix NorESM2.1 (and/or NorESM2.3), however, we also need to add the fix to the [chem preprocessor repo](https://github.com/ESCOMP/CHEM_PREPROCESSOR) used by noresm_develop.

**Comment by annlew** [comment]
Thanks for the comments!

- The issue would apply to all branches where the cam_chempp is present. Are the relevant branches for this update noresm2_1_develop and noresm2_3_develop? I could make pull requests for both. And for other branches/tags users can hopefully find the solution here if they happen to run into this issue. Is it correct that the chem_pp has been separated from cam recently (hadn't found the [chem preprocessor repo](https://github.com/ESCOMP/CHEM_PREPROCESSOR) before), so that future changes to the chempp go into that separate repo?
- I think len=256 should be enough. Most other shorter chars seems to mostly be filenames and short relative paths, however, I wonder if
`character(len=80)  :: iout(100) `
could close to being too short on certain systems. I haven't investigated though.
- Yes, you're right, the path is <root>/components/cam/chem_proc/src/cam_chempp/mozpp.main.f

**Comment by gold2718** [comment]
Yes, please create PRs for noresm2_1_develop and noresm2_3_develop.

**Comment by oyvindseland** [comment]
I have OKed the char length (no reviewers) 
The last test should not bring any problems either? @dirk @gold2718

**Comment by gold2718** [comment]
> The last test should not bring any problems either?

It modifies the source code so we need to adjust it (not trivial)
